src/isomorphic/classic/class/ReactClass.js

```diff
@@ -437,6 +437,19 @@ function validateMethodOverride(isAlreadyDefined, name) {
  */
 function mixSpecIntoComponent(Constructor, spec) {
   if (!spec) {
+    if(__DEV__) {
+      var typeofSpec = typeof spec;
+
+      warning(
+        typeofSpec !== 'object' || typeofSpec === null,
+        '%s: You\'re attempting to ' +
+        'include a mixin that is either null or not an object. ' +
+        'Expected object but got %s',
+        Constructor.displayName || 'ReactClass',
+        typeofSpec === null ? null : typeofSpec
```

Comment:
Why is this ternary operator here?  What value does it add over just `typeofSpec`?